### ISSUE TITLE
Support model parameters

### ISSUE THREAD

**Issue by Leengit** [DigitalSlideArchive/ALBench#47]
Allow the user to override our default parameters for machine learning models, including to select `number of epochs`, `optimizer`, `batch size`, `training weights`, `loss function`, and whether final layer of model is `softmax` vs. `log_softmax`,  In the case of Bayesian machine learning models, user should also be able to override `num_train_samples`, `num_validation_samples`, `num_predict_samples`.
